src/sage/libs/gap/element.pyx

```diff
@@ -77,21 +78,22 @@ cdef Obj make_gap_matrix(sage_list, gap_ring) except NULL:
 
     The list of the elements in ``sage_list`` as a Gap ``Obj``.
     """
-    cdef GapElement l = libgap.eval('[]')
+    cdef GapElement l = GAP_NewPlist(0)
     cdef GapElement elem
     cdef GapElement one
     if gap_ring is not None:
         one = <GapElement>gap_ring.One()
     else:
         one = <GapElement>libgap(1)
-    for x in sage_list:
+    for i in range(sage_list):
```

Comment:
This won't work. I assume you want `for i, x in enumerate(sage_list):`

Reply:
I have fixed that.

Reply:
Thank you